### DISCUSSION TITLE
Übergabe der Token Parameter fehlerhaft

### DISCUSSION THREAD

**Discussion by pbroska** [Digitaler-Impfnachweis/certification-apis#101] [Q&A] [answered]
Guten Tag,
der 5. Schritt (Token) läuft bei mir nicht durch:
curl -H "Content-Type: application/x-www-form-urlencoded" -X POST -d "grant_type=authorization_code&code=70e18e15-b504-4479-956a-0f7028459d8d.346172cc-8911-49db-b069-f3258765d889.d5af5f96-dc70-4402-add3-954386b639f9&redirect_uri=connector://authenticated&client_id=smcb-auth-simulator&session_state=346172cc-8911-49db-b069-f3258765d889" https://keycloak-cert-auth.dev.ibmega.net/auth/realms/master/protocol/openid-connect/token -v
**Fehlermeldung:**
{"error":"invalid_grant","error_description":"Code not valid"}* Connection #0 to host keycloak-cert-auth.dev.ibmega.net left intact

Ich teste im Endpunkt https://keycloak-cert-auth.dev.ibmega.net/auth/realms/master/protocol/openid-connect/auth.

Im Endpunkt der Produktivumgebung: https://id.impfnachweis.info/auth/realms/bmg-ti-certify/protocol/openid-connect/token
Error 400:
{
    "error": "unauthorized_client",
    "error_description": "INVALID_CREDENTIALS: Invalid client credentials"
}
Falsche Zugangsdaten?? session_id+code mehrfach neu generiert.

Hat jemand ein funktionierendes Beispiel für curl?
Vielen Dank schonmal, Patrick

**Comment by ManuelB** [comment, marked as answer]
Ich hab mal unsere Lösung hier detailliert erklärt: https://youtu.be/oLrjF-OISO0 https://github.com/ere-health/dgc-ps-app
